src/write_connection.cpp

```diff
@@ -0,0 +1,41 @@
+#include "write_connection.h"
+
+#define class class_name
+#define private private_ptr
+#include <R_ext/Connections.h>
+#undef class
+#undef private
+
+#if R_CONNECTIONS_VERSION != 1
```

Comment:
It was added in https://github.com/wch/r-source/commit/41b64b12f36851cb87e9132612591d623c217443, and running `git branch --contains 41b64b12f36851cb87e91326125` on the r-source clone returns

```
  R-3-3-branch
  build
  build-trunk
  tags/R-3-0-3
  tags/R-3-1-3
  tags/R-3-2-3
  tags/R-3-2-4
  tags/R-3-2-4-revised
  tags/R-3-2-5
  tags/R-3-3-0
  tags/R-3-3-1
  tags/R-3-3-2
  travis
* trunk
```

So looks like the earliest release was 3.0.3